agent: only remove containers created by SC

tobsan opened the pull request:
Adds a simple regex that checks for container names matching SC-* and if
they don't match - don't remove them.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
agent: only remove containers created by SC

Adds a simple regex that checks for container names matching SC-* and if
they don't match - don't remove them.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

DunderRoffe commented on agent/src/containerutilityinterface.cpp:
```diff
@@ -52,25 +52,29 @@ void ContainerUtilityInterface::removeOldContainers()
-        log_debug() << "Deprecated container " << containerNames[i] << " is successfully destroyed";
+        log_debug() << "Deprecated container " << containerName << " is successfully destroyed";
         delete container;
         delete containerNames[i];
```
Why are we using two lists instead of a map?

tobsan replied:
Because the lists are populated through a call to the LXC API.

DunderRoffe replied:
sure but that does not mean that we need to keep the data stored that way right?

tobsan replied:
No but should we iterate over the data and store it in a different way just because it is convenient?

DunderRoffe replied:
Convenience sounds like a good reason yes :)
Also it would let you use c++11 style for loops

DunderRoffe commented on servicetest/agent/test_agent.py:
```diff
@@ -93,23 +94,41 @@ def test_old_containers_cleanup(self):
+        # Exit and cleanup
         agent_handler.terminate()
-        os.remove("output")
+        os.system("lxc-destroy -n " + skipped_container_name)
```
This cleanup will be skipped it there is an exception (or assertion error) in the code above.

tobsan replied:
The whole test should have a try/finally

DunderRoffe commented on servicetest/agent/test_agent.py:
```diff
@@ -93,23 +94,41 @@ def test_old_containers_cleanup(self):
-        os.remove("output")
+        skipped_container_name = "agent-servicetest"
+        # Create one that SC shouldn't touch
+        os.system("lxc-create -t busybox -n " + skipped_container_name)
```
It might be neater to use the lxc python module for this

tobsan replied:
Working on it. The LXC bindings for Python are Python3 only, which means some refactoring and updating of the service test code.

DunderRoffe reviewed (changes requested):
Looks legit. However, I would like to see the following tests as well:

* Starting more than 1 SC container and 1 non-SC container
* Starting them in different orders
* Starting no SC containers

Commit:
servicetest: use python3

To be able to use the LXC python bindings, we have to use python3.
This commit makes the necessary changes to the service test code.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
ignore the .properties files

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

DunderRoffe commented on servicetest/dbus/dbusapp.py:
```diff
@@ -60,7 +60,7 @@ def Bounce(self, message):
     @dbus.service.method(IFACE, in_signature="s", out_signature="s")
     def Ping(self, message):
-        print "Got a Ping: ", message
+        print(("Got a Ping: ", message))
```
I prefer `print("Got a Ping {}".format(message))` as it explicitly shows where the message will be inserted

DunderRoffe commented on servicetest/dbus/dbusapp.py:
```diff
@@ -153,7 +154,7 @@ def call_ping(self):
-        print "Using mode 'client'"
-        print "method: ", args.method
+        print("Using mode 'client'")
+        print(("method: ", args.method))
```
I prefer `print("method: {}".format(args.method))`

DunderRoffe commented on servicetest/dbus/test_dbus.py:
```diff
@@ -274,9 +275,9 @@ def test_spam_out(self):
-            print("Total time:                   {0:.4f} seconds\n".format(t2 - t0))
+            print(("Clients started:              {0:.4f} seconds\n".format(t1 - t0)))
+            print(("Server received all messages: {0:.4f} seconds\n".format(t2 - t1)))
+            print(("Total time:                   {0:.4f} seconds\n".format(t2 - t0)))
```
Redundant parentheses

DunderRoffe commented on servicetest/filesystem/test_filesystem_strace.py:
```diff
@@ -178,7 +178,7 @@ def test_bindmount_sockets_strace(self):
         # Remove the fileset files from the files list
         for f in fileset:
-            files = filter(lambda a: a != f, files)
+            files = [a for a in files if a != f]
```
A bit weird to mix standard for loop notation with a list comprehension

tobsan replied:
This was suggested by 2to3, actually.

DunderRoffe commented on servicetest/timingprofiling/test_profiling.py:
```diff
@@ -93,7 +93,7 @@ def run_test(num_starts=3):
             """ Start numStarts apps in softwarecontainer.
             """
-            print "Start app " + str(app)
+            print("Start app " + str(app))
```
Use .format instead of +
https://softwareengineering.stackexchange.com/questions/304445/why-is-s-better-than-for-concatenation

DunderRoffe commented on servicetest/timingprofiling/test_profiling.py:
```diff
@@ -134,7 +134,7 @@ def get_python_point(log_file, point_name, match_number=1):
             if match == match_number:
                 return remove_ansi(string.split(line)[2])
-    print "Nothing found! " + point_name + " " + str(match_number)
+    print("Nothing found! " + point_name + " " + str(match_number))
```
Use .format instead of +
https://softwareengineering.stackexchange.com/questions/304445/why-is-s-better-than-for-concatenation

DunderRoffe commented on servicetest/timingprofiling/test_profiling.py:
```diff
@@ -145,9 +145,9 @@ def get_function_log(log_file, function_name, match_number=1):
+                print(line)
                 return remove_ansi(string.split(line)[6])
-    print "Nothing found! " + function_name + " " + str(match_number)
+    print("Nothing found! " + function_name + " " + str(match_number))
```
Use .format instead of +
https://softwareengineering.stackexchange.com/questions/304445/why-is-s-better-than-for-concatenation

DunderRoffe commented on servicetest/timingprofiling/test_profiling.py:
```diff
@@ -165,7 +165,7 @@ def get_log_point(log_file, point_name, match_number=1):
             if match == match_number:
                 return remove_ansi(string.split(line)[4])
-    print "Nothing found! " + point_name + " " + str(match_number)
+    print("Nothing found! " + point_name + " " + str(match_number))
```
Use .format instead of +
https://softwareengineering.stackexchange.com/questions/304445/why-is-s-better-than-for-concatenation

Commit:
bump cookbook

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
service tests: add workaround for certain tests

For some reason, the agent tests make one of the environment tests fail,
so for now we run these tests separately.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
bump cookbook

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

DunderRoffe commented on servicetest/testframework/lib.py:
```diff
@@ -441,7 +441,8 @@ def is_alive(self):
     def terminate(self, kill=False):
-        if self.is_alive():
+        while self.is_alive():
+#        if self.is_alive():
```
Please clean up comment

DunderRoffe commented on servicetest/agent/test_agent.py:
```diff
@@ -133,17 +133,20 @@ def test_old_containers_cleanup(self, containers):
                     container_id = sc.start(DATA)
                     container_name = "SC-" + str(container_id)
-                    sc_containers[container_name] = sc
+                    sc_containers += 1
```
Make sure that sc is cleaned up if there is a failure
